app/domain/authentication/authenticate.rb

```diff
@@ -22,8 +22,8 @@ module Authentication
     def call
       validate_authenticator_exists
```

Comment:
Seems that all the methods are taking authenticator_input as a parameter.
Wouldn't it be better to chain them and pass the authenticator input explicitly?
Should make the code more readable

I would also consider pipelining (|>) the methods to reflect the flow

Reply:
i added a commit so we don't pass around the implicit `authenticator_input` object when it's not necessary. The only place we now pass it over is in the call to `valid?` which is the generic function of each authenticator. we need to pass the whole object there as we don't know which inputs will be used in each authenticator (and it makes sense there to pass an object named `authenticator_input`)